src/main/java/com/slack/kaldb/metadata/core/CorruptMetadataNodeException.java

```diff
@@ -0,0 +1,11 @@
+package com.slack.kaldb.metadata.core;
+
+public class CorruptMetadataNodeException extends RuntimeException {
```

Comment:
We probably don't need a custom exception here - per Effective Java we should favor standard exceptions (#72)  as it makes the API easier to learn, has a smaller memory footprint, and reduces the time loading. An `IllegalStateException` likely makes the most sense here.